### ISSUE TITLE
Ablation study on using just single path encoder?

### ISSUE THREAD

**Issue by lucasjinreal** [luogen1996/LLaVA-HR#1]
What if didn;t use convnext added vision encoder?

**Comment by luogen1996** [comment]
The model you mentioned with a single visual path is exactly LLaVA-1.5. We have conduct extensive comparisons in Tab 1 of our paper. Check out our [paper](https://arxiv.org/pdf/2403.03003.pdf).
![image](https://github.com/luogen1996/LLaVA-HR/assets/38316576/cac412ea-e0bb-454e-9d43-310fd958e136)

**Comment by lucasjinreal** [comment]
I notcied that you enlarge the size in llava-1.5 are using interpolate positional embedding after calculate position_ids.

This would notiablly drop performance as model haven't seen large sizes when training.

What I mean is that, have u did experiment on enlarge input size by interpolate position embedding weight, and then train it along with vision encoder or full model.

How do u think the differences of these two ways.

**Comment by luogen1996** [comment]
I get it, maybe your mentioned way is better. Let me try it.

**Comment by lucasjinreal** [comment]
Nice, let me know the differences between them after you tried.

**Comment by luogen1996** [comment]
I don't see these warnings in my logs.   Weights you print are not used in model, so maybe you can directly ignore them.

**Comment by BlueBlueFF** [comment]
> The model you mentioned with a single visual path is exactly LLaVA-1.5. We have conduct extensive comparisons in Tab 1 of our paper. Check out our [paper](https://arxiv.org/pdf/2403.03003.pdf). ![image](https://private-user-images.githubusercontent.com/38316576/310429347-cac412ea-e0bb-454e-9d43-310fd958e136.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxMjI3MzUsIm5iZiI6MTcxMDEyMjQzNSwicGF0aCI6Ii8zODMxNjU3Ni8zMTA0MjkzNDctY2FjNDEyZWEtZTBiYi00NTRlLTlkNDMtMzEwZmQ5NThlMTM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzExVDAyMDAzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjMDI1ZDA5ZmFkZTdiNjdhODNjYjc0MThlMDhjM2FiNmE3NWU2NGM3ZjU3YzFiZWQ1NDUyNWM0MDZkNGQwZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-LdB6Mm0Lp1wQ3cp21triMSLgNZS1EJ_hFCRIB_998E)

In table 1, are llava 1.5 trained on different resolutions? or only eval in different resolutions?

**Comment by BlueBlueFF** [comment]
> > The model you mentioned with a single visual path is exactly LLaVA-1.5. We have conduct extensive comparisons in Tab 1 of our paper. Check out our [paper](https://arxiv.org/pdf/2403.03003.pdf). ![image](https://private-user-images.githubusercontent.com/38316576/310429347-cac412ea-e0bb-454e-9d43-310fd958e136.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxMjI3MzUsIm5iZiI6MTcxMDEyMjQzNSwicGF0aCI6Ii8zODMxNjU3Ni8zMTA0MjkzNDctY2FjNDEyZWEtZTBiYi00NTRlLTlkNDMtMzEwZmQ5NThlMTM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzExVDAyMDAzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjMDI1ZDA5ZmFkZTdiNjdhODNjYjc0MThlMDhjM2FiNmE3NWU2NGM3ZjU3YzFiZWQ1NDUyNWM0MDZkNGQwZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-LdB6Mm0Lp1wQ3cp21triMSLgNZS1EJ_hFCRIB_998E)
> 
> In table 1, are llava 1.5 trained on different resolutions? or only eval in different resolutions?

@luogen1996 Thanks~

**Comment by luogen1996** [comment]
> > The model you mentioned with a single visual path is exactly LLaVA-1.5. We have conduct extensive comparisons in Tab 1 of our paper. Check out our [paper](https://arxiv.org/pdf/2403.03003.pdf). ![image](https://private-user-images.githubusercontent.com/38316576/310429347-cac412ea-e0bb-454e-9d43-310fd958e136.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxMjI3MzUsIm5iZiI6MTcxMDEyMjQzNSwicGF0aCI6Ii8zODMxNjU3Ni8zMTA0MjkzNDctY2FjNDEyZWEtZTBiYi00NTRlLTlkNDMtMzEwZmQ5NThlMTM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzExVDAyMDAzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjMDI1ZDA5ZmFkZTdiNjdhODNjYjc0MThlMDhjM2FiNmE3NWU2NGM3ZjU3YzFiZWQ1NDUyNWM0MDZkNGQwZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-LdB6Mm0Lp1wQ3cp21triMSLgNZS1EJ_hFCRIB_998E)
> 
> In table 1, are llava 1.5 trained on different resolutions? or only eval in different resolutions?

Yes, we train llava 1.5 on different resolution. The training settings are the same as llava-hr, which includes low-resolution pre-training and high-resolution instruction tuning.

**Comment by CserDu** [comment]
> @luogen1996 Hello, I am doing sft stage2 follow your code, using zero3 finetune, got some warnings:
> 
> ```
> - vision_model.head.layernorm.bias: found shape torch.Size([1152]) in the checkpoint and torch.Size([0]) in the model instantiated
> - vision_model.head.layernorm.weight: found shape torch.Size([1152]) in the checkpoint and torch.Size([0]) in the model instantiated
> - vision_model.head.mlp.fc1.bias: found shape torch.Size([4304]) in the checkpoint and torch.Size([0]) in the model instantiated
> - vision_model.head.mlp.fc1.weight: found shape torch.Size([4304, 1152]) in the checkpoint and torch.Size([0]) in the model instantiated
> - vision_model.head.mlp.fc2.bias: found shape torch.Size([1152]) in the checkpoint and torch.Size([0]) in the model instantiated
> - vision_model.head.mlp.fc2.weight: found shape torch.Size([1152, 4304]) in the checkpoint and torch.Size([0]) in the model instantiated
> - vision_model.head.probe: found shape torch.Size([1, 1, 1152]) in the checkpoint and torch.Size([0]) in the model instantiated
> - vision_model.post_layernorm.bias: found shape torch.Size([1152]) in the checkpoint and torch.Size([0]) in the model instantiated
> - vision_model.post_layernorm.weight: found shape torch.Size([1152]) in the checkpoint and torch.Size([0]) in the model instantiated
> ```
> 
> there was some reports: [microsoft/DeepSpeed#3574](https://github.com/microsoft/DeepSpeed/issues/3574) indicates it related to enabling gradient_checkpoing and zero3 at the same time.
> 
> Does this effect model training? It looks like loss are normal

Hello, I also meet this problem.. Could you please share how you resolved it?
